rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -124,45 +123,41 @@ public synchronized ReentrantLock tryLockTransaction(URI lraId) {
+    }
 
+    public Collection<LRAData> getAll(LRAStatus lraStatus) {
+        if (lraStatus == null) {
+            Collection<LRAData> all = lras.values().stream()
+                    .map(t -> t.getLRAData()).collect(Collectors.toCollection(HashSet::new));
             all.addAll(getAllRecovering());
-
-            return new ArrayList<>(all);
-        }
-
-        if (LRAStatus.Cancelling.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isCancelling).collect(toList());
-        } else if (LRAStatus.Cancelled.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isCancelled).collect(toList());
-        } else if (LRAStatus.FailedToCancel.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isFailedToCancel).collect(toList());
-        } else if (LRAStatus.Closing.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isClosing).collect(toList());
-        } else if (LRAStatus.Closed.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isClosed).collect(toList());
-        } else if (LRAStatus.FailedToClose.name().equals(state)) {
-            return recoveringLRAs.values().stream().map(LRAStatusHolder::new).filter(LRAStatusHolder::isFailedToClose).collect(toList());
+            return all;
         }
 
-        return null;
+        Collection<LRAData> allByStatus = getLraDataByStatus(lras, lraStatus);
+        allByStatus.addAll(getLraDataByStatus(recoveringLRAs, lraStatus));
+        return allByStatus;
     }
 
-    private Set<LRAStatusHolder> getAllActive() {
-        return lras.values().stream().map(LRAStatusHolder::new).collect(toSet());
-    }
-
-    public List<LRAStatusHolder> getAllRecovering(boolean scan) {
+    /**
+     * Getting all the LRA managed by recovery manager. This means all LRAs which are not mapped
+     * only in memory but that were already saved in object store.
+     *
+     * @param scan  defines if there is run recovery manager scanning before returning the collection,
+     *              when the recovery is run then the object store is touched and the returned
+     *              list may be updated with the new loaded objects
+     * @return list of the {@link LRAData} which define the recovering LRAs
+     */
+    public List<LRAData> getAllRecovering(boolean scan) {
         if (scan) {
             RecoveryManager.manager().scan();
         }
 
-        return recoveringLRAs.values().stream().map(LRAStatusHolder::new).collect(toList());
+        return recoveringLRAs.values().stream().map(t -> t.getLRAData()).collect(toList());
     }
 
-    public List<LRAStatusHolder> getAllRecovering() {
+    public List<LRAData> getAllRecovering() {
```

Comment:
we should use unified either Collection or List but I don't see the reason to mix.

Reply:
+1
I would choose `List` since there is a natural ordering for LRAs.

Reply:
+1, good point, I was thinking to use the `Collection` on `getAll` method for I can use the `Set` to prevent duplication of the records in the result collection. Then it happens I introduced this ambiguity. But I can switch back to `List`.